enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,410 @@
+
+The Update Status Controller will run as a part of the hosted control plane. Unlike in Standalone, where it is a CVO operand, USC will be managed by the Control Plane Controller in HCP, where the CVO does not have access to management cluster API and cannot manage resources there. USC needs to access the management cluster API layer to manage the status-reporting API (which targets cluster administrators tasked with updating clusters) and needs to report insights about both the hosted cluster and management cluster resources. It will also need access to the hosted cluster API layer to report on resources present there.
+
+Because `UpdateStatus` API is cluster-scoped, it is not suitable for HCP use, and we will need to introduce a namespaced HCP-specific flavor, such as `HostedClusterUpdateStatus`. HCP-specific API variant will also need to be able to refer to resources in two API surfaces (management and hosted clusters), while the `UpdateStatus` API can rely on all resources being in the same cluster and use simple resource references.
+
+The updates in HCP are not initiated by the [`hcp` client][hcp-client] binary but by [directly modifying `HostedCluster` and `NodePool` resources][updates-in-hcp] using the `oc` client. Hence, at the moment, there is no apparent suitable place to put the update status UX (a client that would consume a `HostedClusterUpdateStatus` instance and present its content). The best place for that is the [`hcp` client][hcp-client], if necessary. It is expected that managed OCP product layers using HCP (ROSA) would consume `HostedClusterUpdateStatus` to improve their update experience, but that work is outside the scope of this enhancement.
+
+When the management cluster is a standalone OCP cluster, there will also be a standard standalone USC instance in the respective platform namespace, managing an `UpdateStatus` that provides information about the management cluster itself.
+
+#### Standalone Clusters
+
+Standalone clusters are the primary target of the functionality and no special considerations are needed.
+
+#### Single-node Deployments or MicroShift
+
+USC and `UpdateStatus` API can work on SNO. USC is updated early in the procedure and will be able to expose the update status when the apiserver is available. USC and `UpdateStatus` can be disabled at installation time through the Capability mechanism for resource-sensitive SNO deployments. Existing SNO clusters using the `vCurrent` capability set would enable USC unless the cluster administrator intervenes by tweaking the capabilities before they update, which may warrant a release note when the feature goes GA.
+
+MicroShift is updated through `rpm-ostree` or `dnf`, so `UpdateStatus` API or any part of USC is not applicable and should not be included in MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+Full API proposal: https://github.com/openshift/api/pull/2012
+
+#### `UpdateStatus` API Overview
```

Comment:
@petr-muller Is this example up to date with everything in the openshift/api PR? Any chance this could be made a bit more complete with the full API represented?

Reply:
By that I mean, when I look at the API PR it seems like there's a lot more there than illustrated here, can this be filled in to illustrate a more complete UpdateStatus resource?

Reply:
I'll come up with up to date and further populated example of the API

Reply:
Example input + output + walkthrough added here:

https://github.com/openshift/enhancements/blob/7687d60346060818d7f4725ef0d4529ca197b26b/enhancements/update/update-status-api/update-status-api-example.md